### PR TITLE
Fx UI  v1.0

### PR DESCRIPTION
**Pull request by utsavkumar-280**
This is the first iteration of my css component library.I have tried to made it somewhat responsive,but I think there is still room for improvement. Please provide your valuable feedbacks.
Here is the [live demo](https://fx-ui.netlify.app/) of the documentaion site.

### CODE REVIEW

File: ComponentsStyles/root.css [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  --primaryBgLight: #ffffff;
+  --primaryBgDark: #f1f1f1;
+  --divider: #c2c2c25d;
+  --primaryScroll: #d1d1d1;
+
+  --hover: #28beac;
+
+  --secondary: #0e7490;
+  --secondaryLight: #e2feff;
+  --secondary1: #2dd4bf;
```
Comment (pruthvirajmv) [inline ComponentsStyles/root.css]: could you please be descriptive instead of 1 in --secondary1

File: ComponentsStyles/root.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  --thirdPrimary: #14b8a6;
+  --thirdPrimary1: #2dd4bf;
```
Comment (pruthvirajmv) [inline ComponentsStyles/root.css]: could you please be descriptive instead of 1 in --thirdPrimary1

File: doc.css [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+  padding: 2rem;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  flex-wrap: wrap;
+}
+.alignCenter {
+  align-items: center;
+}
+#vh50 {
```
Comment (pruthvirajmv) [inline doc.css]: Not a good practice to use the IDs as CSS selectors to style
Reason IDs should be unique on a page, override class-based styles, specificity issues, no possibility of reuse

**Comment by pruthvirajmv** [comment]
Hello Utsav,
CSS styles are great and clean code is observed throughout the codebase.
above are few recommendations from my side and below are few more points in addition to that,

**Comment by pruthvirajmv** [comment]
> Hello Utsav,
> CSS styles are great and clean code is observed throughout the codebase.
> above are few recommendations from my side and below are few more points in addition to that,

File folder structure - all your components can be in  a single folder like there is a for content & forms

**Comment by pruthvirajmv** [comment]
File folder structure - all your components can be in  a single folder like there is a for content & forms

**Comment by pruthvirajmv** [comment]
UI
1. navbar in doc site for mobile screens breaking as seen below
image

2. instead of showing the js logic part and copy option, can think of asking the user to include jslogic file in the script tag of their HTML

**Comment by utsavkumar-280** [comment]
It only does that in inspect element sometimes .On my phone it's working properly .After resizing screen size a bit ,it starts working fine

**Comment by Sachanh** [comment]
As Pruthvi Mentioned for UI Part  the content is coming over the navbar. To remove this things use z-index of 2-3 to fix this.

image

Z-index of 2 fixes this

image

**Comment by Sachanh** [comment]
As checked you have used viewport unit vw for setting the viewport. 
I suggest you to use % for setting the width instead viewport unit. As this units still has some bugs in certain browsers, which affects the website responsiveness.

to get an overview of it Read this article: https://www.sitepoint.com/css-viewport-units-quick-start/

image

**Comment by Sachanh** [comment]
Commented out things can be removed .
image
image

**Comment by sakshijainn** [comment]
Hi Utsav: 
1. Your div.header is overlappig with div.main due to which when I choose any component, the content gets hidden
For Reference:
image


2. In mobile version, when I open the hamburger menu : and choose the component , do apply some event listener such that when I choose the component , the components navigation bar closes automatically, it will enhance the UI 
For Reference:
image

**Comment by sakshijainn** [comment]
Thankyou for refactoring I think you can now merge into main branch

**Comment by utsavkumar-280** [comment]
All clear, merging now

**Pull request merged**